code/game/gamemodes/miniantags/revenant/revenant_abilities.dm

```diff
@@ -23,7 +23,11 @@
 		return
 	A.attack_ghost(src)
 	if(ishuman(A) && in_range(src, A))
-		Harvest(A)
+		if(isLivingSSD(A))
+			if(client && client.send_ssd_warning(A)) //Do NOT Harvest(SSD people)
+				return
+		else
+			Harvest(A)
```

Comment:
```suggestion
		if(isLivingSSD(A) && client && client.send_ssd_warning(A)) //Do NOT Harvest SSD people unless you accept the warning
			return
		Harvest(A)
```


The issue was with your `if` `else`. Even though `send_ssd_warning()` is properly used here the else bit of the code is never called, as if the person is SSD it always calls the first bit.
The check above just condenses what you did in a more readable format, and if the target is SSD _but_ the user has acknowledged the ssd warning, they can still harvest.

Reply:
Ah, now that makes sense!

I didn't understand why the send_ssd_warning() method was returning a boolean, it being the user acknowledgment is perfectly logical! Yeah, if it returns true they should be able to harvest the SSD player because they were authorized.

Thanks!

Reply:
Counterintuitively, `send_ssd_warning()` returns FALSE when the user has already acknowledged. I'm not overly sure why it was done this way but a TRUE return value means they have not acknowledged, hence the return.